webapp/graphite/render/grammar_unsafe.py

```diff
@@ -93,7 +105,8 @@ def setRaw(s, loc, toks):
 ).setParseAction(setRaw)('call')
 
 # Metric pattern (aka. pathExpression)
-validMetricChars = ''.join((set(printables) - set(symbols)))
+unicodePrintables = u''.join(unichr(c) for c in xrange(sys.maxunicode) if not unichr(c).isspace())
```

Comment:
2 concerns here:
1. This is a huge range
2. Should we be excluding more characters than just spaces, there are many unicode categories https://en.wikipedia.org/wiki/Unicode_character_property#General_Category

Reply:
1. Yes, I missed that point, thanks! I'll probably put it behind switch then - if user want UTF-8 in metrics, it can enable that. Would it be better?
2. I think only spaces is good for start, not sure should we add anything later

Reply:
Agree with Dan, this is basically the reason why I didn't create the PR back then, I didn't think this was an elegant solution and didn't know how to do it properly.

Reply:
Oh, TIL pyparsing 2.3.0 and newer giving you `pyparsing_unicode.printables` !
But it's better to hide it behind switch anyway

Reply:
Ah `pyparsing_unicode.printables` is much better! Still not sure if this is the right way to do this but as long it is behind the switch it is good enough for me.